src/modal/modal.ts

```diff
@@ -48,24 +59,35 @@ export class CaptureEyeModal extends LitElement {
     if (setAsLoaded) this._assetLoaded = true;
   }
 
-  updateEngagementZone(
-    engagementImage: string,
-    engagementLink: string,
-    actionButtonText: string,
-    actionButtonLink: string
-  ) {
-    if (this.engagementImage !== engagementImage) {
-      this._imageLoaded = false;
-      this.engagementImage = engagementImage;
-      this.requestUpdate();
+  updateModalOptions(options: ModalOptions) {
+    if (options.nid) this.nid = options.nid;
+    if (options.layout) this.layout = options.layout;
+    if (options.copyrightZoneTitle)
+      this._copyrightZoneTitle = options.copyrightZoneTitle;
+    if (options.engagementImage)
+      this._engagementImage = options.engagementImage;
```

Comment:
Just to confirm, does _imageLoaded need to be cleared? The previous method reset the value when the engagementImage changed, but now there's no place where the value is being reset.

Reply:
Yes I believe it should be cleared as well. Let me check and update it.

Reply:
I couldn’t find where it was cleared. Could you point it out for me? Thanks!

Reply:
@olgahaha Updated in the latest commit